### PR TITLE
Add integration test for XACML multi-decision profile in JSON format

### PR DESCRIPTION
**Pull request by GDRDABARERA**
An integration test case to test XACML multi-decision profile flow

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.Set;
+
```
Comment (nilasini) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java]: Shall we add a class level comment?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+        super.init();
+        entitlementPolicyClient = new EntitlementPolicyServiceClient(backendURL, sessionCookie);
+        entitlementServiceClient = new EntitlementServiceClient(backendURL, sessionCookie);
+        remoteUserStoreManagerServiceClient = new RemoteUserStoreManagerServiceClient(backendURL, sessionCookie);
+        PolicyDTO policy = new PolicyDTO();
+        policy.setPolicy(POLICY);
+        entitlementPolicyClient.addPolicy(policy);
+        entitlementPolicyClient.publishPolicies(new String[]{POLICY_ID}, new String[]{"PDP Subscriber"}, "CREATE",
+                true, null, 1);
+        remoteUserStoreManagerServiceClient.addUser("adminUser","abc123", new String[]{"admin"}, null,
```
Comment (nilasini) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java]: Assign the "abc123" & "admin" to a constant and use it

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+
+        String webRespose = client.post(request, String.class);
+        JSONObject objReturn = new JSONObject(webRespose);
+        Assert.assertTrue(areEqual(objExpected, objReturn), "The response is wrong it should be :"+ response + " But" +
+                " the response is :" + webRespose);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void destroy() throws Exception {
+        //Remove added policy
```
Comment (nilasini) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java]: Add a space before starting to write the comment

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+            }
+
+            scanner.close();
+
+        } catch (IOException e) {
+            log.error("IO Exception in reading test case");
+        } catch (URISyntaxException e) {
+            log.error("IO Exception in reading test case");
+        }
+
```
Comment (nilasini) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java]: remove extra spaces

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+        return result.toString().replaceAll("\\n\\r|\\n|\\r|\\t|\\s{2,}", "").replaceAll(": ", ":");
+    }
+
```
Comment (nilasini) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java]: Remove extra lines

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                result.append(line).append("\n");
+            }
+
+            scanner.close();
+
+        } catch (IOException e) {
+            log.error("IO Exception in reading test case");
```
Comment (nilasini) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java]: IO exception in reading test case? Is it correct?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+        entitlementPolicyClient = new EntitlementPolicyServiceClient(backendURL, sessionCookie);
+        remoteUserStoreManagerServiceClient = new RemoteUserStoreManagerServiceClient(backendURL, sessionCookie);
+        entitlementPolicyClient.enableDisablePolicy(POLICY_ID, false);
+        entitlementPolicyClient.removePolicy(POLICY_ID);
+        remoteUserStoreManagerServiceClient.deleteUser("adminUser");
+        remoteUserStoreManagerServiceClient.deleteUser("publicUser");
+        remoteUserStoreManagerServiceClient.deleteUser("localUser");
+    }
+
+    public static boolean areEqual(Object ob1, Object ob2) throws JSONException {
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java]: shall we have this like areEqualJsonObjects?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java (hunk 6 above)
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java]: pass 'e' as a param

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+                String line = scanner.nextLine();
+                result.append(line).append("\n");
+            }
+
+            scanner.close();
+
+        } catch (IOException e) {
+            log.error("IO Exception in reading test case");
+        } catch (URISyntaxException e) {
+            log.error("IO Exception in reading test case");
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java]: pass 'e' as a param, and consider changing the message to a meaningful one

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+        WebClient client = WebClient.create(ENDPOINT_ADDRESS);
+
+        client.header("Authorization", "Basic YWRtaW46YWRtaW4=");
+        client.type("application/json");
+        client.accept("application/json");
+
+        client.path("pdp");
+
+        String request = readReource("entitlement/json/request2.json");
+        String response = readReource("entitlement/json/response2.json");
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java]: shall we have some meaningful values to the files also?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java (hunk 1 above)
Comment (GDRDABARERA) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java]: Added

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+
+            File file = new File(filepath);
+
+            Scanner scanner = new Scanner(file);
+
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                result.append(line).append("\n");
+            }
+            scanner.close();
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java]: do this in the 'finally' block

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java [hunk 11]
```diff
@@ -0,0 +1,312 @@
...
+            }
+            return jsonSet;
+        } else {
+            return elem;
+        }
+    }
+
+    private String readReource(String path) {
+
+        StringBuilder result = new StringBuilder("");
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementJSONSupportMultiDecisionProfileTestCase.java]: empty string is redundant here

**Pull request merged**